common/src/main/java/com/scottlogic/deg/common/profile/DateTimeGranularity.java

```diff
@@ -24,6 +24,12 @@ public DateTimeGranularity(ChronoUnit chronoUnit, boolean workingDay) {
         this.temporalAdjusterGenerator = new TemporalAdjusterGenerator(chronoUnit, workingDay);
     }
 
+    public static DateTimeGranularity create(String granularity){
+        String offsetUnitUpperCase = granularity.toUpperCase();
+        boolean workingDay = offsetUnitUpperCase.equals("WORKING DAYS");
+        return new DateTimeGranularity(Enum.valueOf(ChronoUnit.class, workingDay ? "DAYS" : offsetUnitUpperCase), workingDay);
+    }
+
```

Comment:
Why have we gone for this create pattern?

Reply:
It provides more flexibility in validating as it has a return type which a constructor will not have. If you wanted to return either an error object or a valid object through an either or an option you wouldn't be able to do this in a constructor.